sequencers/modeshape-sequencer-audio/src/main/java/org/modeshape/sequencer/audio/AudioMetadataSequencer.java

```diff
@@ -0,0 +1,155 @@
+            outputNode.setPrimaryType(AudioMetadataLexicon.METADATA_NODE);
+        } else {
+            sequencedNode = outputNode.addNode(AudioMetadataLexicon.METADATA_NODE, AudioMetadataLexicon.METADATA_NODE);
+        }
+
+        sequencedNode.setProperty(AudioMetadataLexicon.FORMAT_NAME, metadata.getFormatName());
+        setPropertyIfMetadataPresent(sequencedNode, JcrConstants.JCR_MIME_TYPE, mimeType);
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.BITRATE, metadata.getBitrate());
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.SAMPLE_RATE, metadata.getSampleRate());
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.CHANNELS, metadata.getChannels());
+        setPropertyIfMetadataPresent(sequencedNode, AudioMetadataLexicon.LENGTH, metadata.getLength());
+
+        addTagNode(sequencedNode, metadata);
+        return true;
+    }
+
+    private void addTagNode( Node sequencedNode,
+                             AudioMetadata metadata ) throws ItemExistsException, PathNotFoundException, NoSuchNodeTypeException, LockException, VersionException, ConstraintViolationException, RepositoryException {
+        Node tagNode = sequencedNode.addNode(AudioMetadataLexicon.TAG_NODE, AudioMetadataLexicon.TAG_NODE);
+
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.TITLE, metadata.getTitle());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.ARTIST, metadata.getArtist());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.ALBUM, metadata.getAlbum());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.YEAR, metadata.getYear());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.COMMENT, metadata.getComment());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.TRACK, metadata.getTrack());
+        setPropertyIfMetadataPresent(tagNode, AudioMetadataLexicon.GENRE, metadata.getGenre());
+
+        for (AudioMetadataArtwork artwork : metadata.getArtwork()) {
+            Node artworkNode = tagNode.addNode(AudioMetadataLexicon.ARTWORK_NODE, AudioMetadataLexicon.ARTWORK_NODE);
+            setPropertyIfMetadataPresent(artworkNode, JcrConstants.JCR_MIME_TYPE, artwork.getMimeType());
+            setPropertyIfMetadataPresent(artworkNode, AudioMetadataLexicon.ARTWORK_TYPE, artwork.getType());
+            setPropertyIfMetadataPresent(artworkNode, JcrConstants.JCR_DATA, artwork.getData());
+        }
+    }
+
+    private void setPropertyIfMetadataPresent( Node node,
+                                               String propertyName,
+                                               Object value) throws RepositoryException {
+        if (value != null) {
+            if (value instanceof String && !StringUtil.isBlank((String) value)) {
+                node.setProperty(propertyName, (String) value);
+            }
+            if (value instanceof Long) {
```

Comment:
if the value here can never be a floating point value (Double or Float) you can change the 2 instanceofs to something along the lines of:
```
if (value instanceof Number) {
   setProperty(name, ((Number) value).longValue())
}